### PR TITLE
Add ability to configure localstack image used

### PR DESCRIPTION
**Pull request by rushton**
Some organizations restrict use of public images for security and
network performance. This commit supports changing the default
image to something other than the public localstack image.

### CODE REVIEW

File: localstack.go [hunk 1]
```diff
@@ -214,7 +223,7 @@ func (i *Instance) start(ctx context.Context) error {
 }
 
 func (i *Instance) startLocalstack(ctx context.Context) error {
-	imageName := "localstack/localstack:" + i.version
+	imageName := i.image + i.version
```
Comment (rushton) [inline localstack.go:226]: there exist some tests which bypass the NewInstance which cause a nil pointer panic. Not sure on the best course of action here, could use an opinion as I'm still somewhat new to golang.

Comment (elgohr) [reply localstack.go:226]: Why don't you use a Pull Through Registry (https://docs.docker.com/registry/recipes/mirror/)? There's no guarantee that every docker image will work the same as localstack.

**Comment by elgohr** [comment]
Why would those organizations trust me (and other contributors of this library) more than the contributors of localstack?
Regarding performance a pull-through registry can be used.
Of course these companies would be free to fork this repository.
I'm not taking the extra effort to make this library resilient to non-known image behaviour.

**Comment by rushton** [comment]
Some organizations do not allow pulling public docker images in any capacity, they first vet images, then publish to their internal registry. Regarding compatibility, it would be up to the caller to ensure the image they're using is compatible.

For others with the same need, I've found [testcontainers-go](https://github.com/testcontainers/testcontainers-go) provides the functionality I need with minor setup for localstack.

**Pull request closed without merging**